jsapp/js/account/usage/usage.component.tsx

```diff
@@ -121,6 +138,56 @@ export default function Usage() {
     getLimits();
   }, [productsContext.isLoaded, oneTimeAddOnsContext.isLoaded]);
 
+  function filterAddOns(type: USAGE_TYPE) {
+    const availableAddons = oneTimeAddOnsContext.addons.filter(
+      (addon) => addon.is_available
+    );
+    
+    // Find the relevant addons, but first check and make sure add-on
+    // limits aren't superceded by an "unlimited" usage limit.
+    switch (type) {
+      case USAGE_TYPE.SUBMISSIONS:
+        return limits.submissionsRecurringLimit !== Limits.unlimited
+          ? availableAddons.filter(
+              (addon) => addon.total_usage_limits.submission_limit
+            )
+          : [];
+      case USAGE_TYPE.TRANSCRIPTION:
+        return limits.nlpMinuteRecurringLimit !== Limits.unlimited
+          ? availableAddons.filter(
+              (addon) => addon.total_usage_limits.asr_seconds_limit
+            )
+          : [];
+      case USAGE_TYPE.TRANSLATION:
+        return limits.nlpCharacterRecurringLimit !== Limits.unlimited
+          ? availableAddons.filter(
+              (addon) => addon.total_usage_limits.mt_characters_limit
+            )
+          : [];
+      default:
+        return [];
+    }
+  }
+
+  // Find out if any usage type has one-time addons so we can
+  // adjust the formatting of the usage containers to accommodate
+  // a detail link.
+  const useAddonsLayout = useMemo(() => {
```

Comment:
Because we're using a lot of react hooks, `useAddonsLayout` should get renamed to avoid confusion. [Hooks always start with `use`](https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use), and this is a boolean (although it's generated by a hook.)